mage_ai/orchestration/monitor/monitor_stats.py

```diff
@@ -63,15 +66,23 @@ def get_pipeline_run_count(
         stats_by_schedule_id = dict()
         pipeline_type_by_pipeline_uuid = dict()
         for p in pipeline_runs:
-            if p.pipeline_schedule is None:
+            if p.pipeline_schedule is None and not group_by_pipeline_type:
                 continue
-            if p.pipeline_schedule_id not in stats_by_schedule_id:
-                stats_by_schedule_id[p.pipeline_schedule_id] = dict(
-                    name=p.pipeline_schedule_name,
+            if p.pipeline_schedule_id is None:
+                pipeline_schedule_id = NO_PIPELINE_SCHEDULE_ID
+                pipeline_schedule_name = NO_PIPELINE_SCHEDULE_NAME
+            else:
+                pipeline_schedule_id = p.pipeline_schedule_id
+                pipeline_schedule_name = p.pipeline_schedule_name
+            if pipeline_schedule_id not in stats_by_schedule_id:
+                stats_by_schedule_id[pipeline_schedule_id] = dict(
+                    name=pipeline_schedule_name,
                     data=dict(),
                 )
             created_at_formatted = p.created_at.strftime('%Y-%m-%d')
-            data = stats_by_schedule_id[p.pipeline_schedule_id]['data']
+            data = stats_by_schedule_id[p.pipeline_schedule_id]['data'] \
```

Comment:
again, you can simplify the code here `data = stats_by_schedule_id[pipeline_schedule_id]['data']`